File: node-child_process/hello-world-with-args.js
// Import the child_process module
import { exec } from 'child_process';

const args = process.argv.slice(2); //because arg-0 is path to the Node.js executable and arg-1 is path to this script file

// Command to run (example: list files)
exec(`dotnet run --file hello-world-with-args.cs -- ${args[0]}`,
    (error, stdout, stderr) => {
        if (error) {
            console.error(`Error executing command: ${error.message}`, {error});

            return;
        }

        if (stderr) {
            console.error(`Command error output: ${stderr}`, {stderr});

            return;
        }

        console.log(`Command output:\n${stdout}`);
    });

File: node-child_process/hello-world-with-args.cs
if (args.Length > 0)
{
    switch (args[0])
    {
        case "do_err":
            TextWriter errorWriter = Console.Error;

            errorWriter.WriteLine("Doing error!");

            return -1;

        default:
            Console.WriteLine($"Hello {args[0]}!");

            return 0;
    }
}
else
    Console.WriteLine("Hello world!");

    return 0;
